src/placeos-frontend-loader/loader.cr

```diff
@@ -32,10 +39,33 @@ module PlaceOS::FrontendLoader
     getter update_crontab : String
     private property update_cron : Tasker::CRON(Int64)? = nil
 
+    def self.get_actioner(ref : Remote::Reference) : Remote
+      case ref.remote_type
+      in Remote::Reference::Type::Github
+        PlaceOS::FrontendLoader::Github.new
+      in Remote::Reference::Type::GitLab
+        PlaceOS::FrontendLoader::GitLab.new
+      in Nil
+        raise Exception.new("repository uri not supported")
+      end
+    end
+
+    def set_actioner(remote_type : String)
+      case Remote::Reference::Type.parse?(remote_type)
+      in Remote::Reference::Type::Github
+        @actioner = PlaceOS::FrontendLoader::Github.new
+      in Remote::Reference::Type::GitLab
+        @actioner = PlaceOS::FrontendLoader::GitLab.new
+      in Nil
+        raise Exception.new("repository uri not supported")
+      end
+    end
```

Comment:
Why the single `actioner`?
It looks like you'll have to recreate the object and ensure access is synchronised if you do it this way.
Perhaps a lazy hash of `Hash(Type, Remote)` could work here?

Another small suggestion would be having a single method that goes from `Type` to `Remote`, and making it perform that transformation via macro. Main reason for suggesting is that this method is essentially the same as `get_actioner`. 

`actioner` could be renamed to `remote` to match the abstract class name, to reduce a layer of indirection.